lms/djangoapps/courseware/field_overrides.py

```diff
@@ -70,6 +69,12 @@ def wrap(cls, user, wrapped):
         any performance impact of this feature if no override providers are
         configured.
         """
+        # if ccx not enabled for this course, just return unwrapped
+        from courseware.courses import get_course_by_id
+        course = get_course_by_id(course_id)
+        if not course.enable_ccx:
+            return wrapped
+        
         if cls.provider_classes is None:
```

Comment:
@cguardia, this feels to me like the right place to do the check.  Rather than above.  If the provider class provides an API that allows checking it against a course, we could rewrite the list comprehension that follows here as:

```python

provider_classes = (resolve_dotted(name) for name in settings.FIELD_OVERRIDE_PROVIDERS)
cls.provider_classes = tuple((klass for klass in provider_classes if klass.enabled_for_course(course_id)))
```

The problem I'm starting to see, though, is that the list of available provider classes is currently a class attribute, and once set it is not looked up again.  That is going to have to change to allow the calculation of available providers to be recalculated each time a call to `OverrideFieldData.wrap()` is made.

@cpennington because of the trade of a one-time run-time monkey patch of the class object for a check that happens each time the wrap method is called, are we trading one problem for another?  Will the increased number of checks for available providers cancel out the benefit of not wrapping for courses that have an override provider disabled?